examples/hybrid/TurbulenceConvectionUtils.jl

```diff
@@ -102,31 +101,81 @@ function init_tc!(Y, p, params, colidx)
 
     FT = eltype(edmf)
     # `nothing` goes into State because OrdinaryDiffEq.jl owns tendencies.
     state = TC.tc_column_state(Y, p, nothing, colidx)
-    thermo_params = CAP.thermodynamics_params(params)
-
     grid = TC.Grid(state)
-    FT = CC.Spaces.undertype(axes(Y.c))
-    C123 = CCG.Covariant123Vector
-    t = FT(0)
 
-    # TODO: can we simply remove this?
-    If = CCO.InterpolateC2F(bottom = CCO.Extrapolate(), top = CCO.Extrapolate())
-    @. p.edmf_cache.aux.face.ρ[colidx] = If(Y.c.ρ[colidx])
+    # parent(state.aux.face) .= NaN
+    # parent(state.aux.cent) .= NaN
 
-    # TODO: convert initialize_profiles to set prognostic state, not aux state
+    # Initialization of auxiliary variables:
+    (; ᶜp, ᶜts) = p
+    ᶜθ_liq_ice = p.edmf_cache.aux.cent.θ_liq_ice
+    ᶜq_tot = p.edmf_cache.aux.cent.q_tot # only initialized for moist profiles
+    ᶜtke = p.edmf_cache.aux.cent.tke
+    ᶜHvar = p.edmf_cache.aux.cent.Hvar # only initialized for GABLS
+    ᶜHvar .= 0
     Cases.initialize_profiles(case, grid, thermo_params, state)
+    if p.atmos.moisture_model isa CA.DryModel
+        @. ᶜts[colidx] =
+            TD.PhaseDry_pθ(thermo_params, ᶜp[colidx], ᶜθ_liq_ice[colidx])
+    else
+        @. ᶜts[colidx] = TD.PhaseEquil_pθq(
+            thermo_params,
+            ᶜp[colidx],
+            ᶜθ_liq_ice[colidx],
+            ᶜq_tot[colidx],
+        )
+    end
 
-    # Temporarily, we'll re-populate ρq_tot based on initial aux q_tot
-    q_tot = p.edmf_cache.aux.cent.q_tot[colidx]
-    @. Y.c.ρq_tot[colidx] = Y.c.ρ[colidx] * q_tot
-    set_thermo_state_pθq!(Y, p, colidx)
-    set_grid_mean_from_thermo_state!(thermo_params, state, grid)
-    assign_thermo_aux!(state, grid, edmf.moisture_model, thermo_params)
-    initialize_edmf(edmf, grid, state, surf_params, tc_params, t)
+    # Initialization of prognostic variables:
+    (; ᶜinterp) = p.operators
+    C123 = CCG.Covariant123Vector
+    @. Y.c.ρ[colidx] = TD.air_density(thermo_params, ᶜts[colidx])
+    @. Y.c.ρe_tot[colidx] =
+        Y.c.ρ[colidx] * TD.total_energy(
+            thermo_params,
+            ᶜts[colidx],
+            LA.norm_sqr(C123(Y.c.uₕ[colidx]) + C123(ᶜinterp(Y.f.w[colidx]))) /
+            2,
+            p.ᶜΦ[colidx],
+        )
+    if !(p.atmos.moisture_model isa CA.DryModel)
+        @. Y.c.ρq_tot[colidx] = Y.c.ρ[colidx] * ᶜq_tot[colidx]
+    end
+
+    N_up = TC.n_updrafts(edmf)
+    a_min = edmf.minimum_area
+    lg = CC.Fields.local_geometry_field(axes(Y.f))
+    @inbounds for i in 1:N_up
+        @. Y.c.turbconv.up[i].ρarea[colidx] = Y.c.ρ[colidx] * a_min
+        @. Y.c.turbconv.up[i].ρaq_tot[colidx] =
+            Y.c.turbconv.up[i].ρarea[colidx] * ᶜq_tot[colidx]
+        @. Y.c.turbconv.up[i].ρaθ_liq_ice[colidx] =
+            Y.c.turbconv.up[i].ρarea[colidx] * ᶜθ_liq_ice[colidx]
+        @. Y.f.turbconv.up[i].w[colidx] =
+            CC.Geometry.Covariant3Vector(CC.Geometry.WVector(FT(0)), lg[colidx])
```

Comment:
IN the deleted code we also had this hack:
```
        a_surf = TC.area_surface_bc(surf, edmf, i)
        aux_up[i].area[kc_surf] = a_surf
        prog_up[i].ρarea[kc_surf] = ρ_c[kc_surf] * a_surf
```

Do you think this could be the reason why the results are changed here?

Reply:
This operation is already being performed in `set_edmf_surface_bc`.